src/NativeAdComponents.js

```diff
@@ -104,7 +104,8 @@ export const IconView = (props) => {
   if (!nativeAdView) return null;
 
   return (
-    <Image {...props} ref={imageRef} source={{uri: nativeAd.url || null}}/>
+    nativeAd?.url ? <Image {...props} source={{uri: nativeAd.url}} /> :
+      nativeAd?.image ? <Image {...props} ref={imageRef} /> : <Image {...props} />
```

Comment:
nit: for clarity can we also break out the last statement
```suggestion
      nativeAd?.image ? <Image {...props} ref={imageRef} /> : 
      <Image {...props} />
```